types/jasmine/jasmine-tests.ts

```diff
@@ -1990,6 +1990,17 @@ describe("better typed spys", () => {
             spyOn<Base>(new Super(), "service2"); // $ExpectError
         });
     });
+    describe("spyOnProperty", () => {
+        it("works", () => {
+            const obj = {prop: "test", otherProp: 1};
+            const getSpy = spyOnProperty(obj, "prop");
+            getSpy.and.returnValue("spy");
+            getSpy.and.returnValue(123); // $ExpectError
+            const setSpy = spyOnProperty(obj, "prop", "set");
+            setSpy.calls.first().args; // $ExpectType [string] || [value: string]
```

Comment:
What exactly is the type on L2000? It doesn't make much sense to me

I get [string] is a 1 element tuple, but [value: string] I don't know?

Reply:
Tuples in TypeScript may have labels: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#labeled-tuple-elements.

There is another example in L1984.